pkgs/os-specific/linux/kernel/common-config.nix

```diff
@@ -213,6 +213,11 @@ let
       MPTCP           = whenAtLeast "5.6" yes;
       MPTCP_IPV6      = whenAtLeast "5.6" yes;
       INET_MPTCP_DIAG = whenAtLeast "5.9" (mkDefault module);
+
+      # Kernel TLS
+      TLS         = whenAtLeast "4.13" yes;
+      TLS_DEVICE  = whenAtLeast "4.18" yes;
+      TLS_TOE     = whenAtLeast "5.5" yes;
```

Comment:
Not sure about this one. It's described as "Enable kernel support for legacy HW offload of the TLS protocol,
which is incompatible with the Linux networking stack semantics." and idk if that's something we want.

The other two are probably fine.

Reply:
What do other distros do?

Reply:
Ubuntu does this:
```
CONFIG_TLS                                      policy<{'amd64': 'm', 'arm64': 'm', 'armhf': 'm', 'ppc64el': 'm', 's390x': 'm'}>
CONFIG_TLS_DEVICE                               policy<{'amd64': 'y', 'arm64': 'y', 'armhf': 'y', 'ppc64el': 'y', 's390x': 'y'}>
CONFIG_TLS_TOE                                  policy<{'amd64': 'n', 'arm64': 'n', 'armhf': 'n', 'ppc64el': 'n', 's390x': 'n'}>
```

And says in their changelog:
```
    - [Config] annotations: CRYPTO_DEV_CHELSIO_TLS depends on legacy (and largely
      unmantained) TLS_TOE
```

Fedora (and RHEL) don't enable it either, see https://src.fedoraproject.org/rpms/kernel/blob/rawhide/f/kernel-x86_64-fedora.config and https://src.fedoraproject.org/rpms/kernel/blob/rawhide/f/kernel-x86_64-rhel.config.

So my vote would be `TLS` as module, `TLS_DEVICE` yes and no `TLS_TOE`.

Reply:
Ok, removed TLS_TOE.